grafana/risingwave-dashboard.dashboard.py

```diff
@@ -1726,6 +1738,11 @@ def section_hummock(panels):
                     f"(sum(rate({metric('file_cache_miss')}[$__rate_interval])) by (instance)) / (sum(rate({metric('file_cache_latency_count', file_cache_get_filter)}[$__rate_interval])) by (instance))",
                     "file cache miss rate @ {{instance}}",
                 ),
+
+                panels.target(
+                    f"((sum(rate({metric('state_store_read_req_non_exist_counts')}[$__rate_interval])) by (job,instance,table_id,type)) - (sum(rate({metric('state_store_read_req_check_bloom_filter_counts')}[$__rate_interval])) by (job,instance,table_id,type)) + (sum(rate({metric('state_store_read_req_bloom_filter_hit_counts')}[$__rate_interval])) by (job,instance,table_id,type))) / (sum(rate({metric('state_store_read_req_check_bloom_filter_counts')}[$__rate_interval])) by (job,instance,table_id,type))",
+                    "read req bloom filter false positive - {{table_id}} - {{type}} @ {{job}} @ {{instance}}",
+                ),
```

Comment:
This looks incorrect.
`state_store_read_req_non_exist_counts` records the count of non-existence key read. (`A`)
`state_store_read_req_check_bloom_filter_counts` records the count of read that checks SST bloom filters. (`B`)
`state_store_read_req_bloom_filter_hit_counts` records the count of read that checks SST bloom filters and at least one bloom filter returns true. (`C`)

We cannot derive read false positive rate from these metrics. Instead, we should change `A` to record the count of read **that checks SST bloom filters** but returns nothing (read on non-existence key).

In this case:
- `bloom filters all false count = B - C`
- `bloom filter filter rate = (B - C) / B`
- `true positive count = B - A`
- `true positive rate = (B - A) / C`
- `false positive rate = 1 - true positive rate = (C - B + A) / C`

Reply:
Can we use `state_store_read_req_bloom_filter_positive_counts` instead of `state_store_read_req_bloom_filter_hit_counts`. Hit here sound a little bit confusing.

Reply:
> Can we use `state_store_read_req_bloom_filter_positive_counts` instead of `state_store_read_req_bloom_filter_hit_counts`. Hit here sound a little bit confusing.

+1

Reply:
After refactoring, we record the count of read that checks SST bloom filters (at least one return true) but returns nothing as A, so we can adjust the formula to

false positive rate = 1 - true positive rate =  1 - (A / C)

Reply:
@hzxa21  For this PR, we only calculate the FPR of the request of `state_store`. But for one get request, we may fetch several sst files, and if we only get result in the last file, the request would be record as `true positive` because it does not return None, while is_non_exist_key is false.

Reply:
> @hzxa21 For this PR, we only calculate the FPR of the request of `state_store`. But for one get request, we may fetch several sst files, and if we only get result in the last file, the request would be record as `true positive` because it does not return None, while is_non_exist_key is false.

Yes, we are aware of that and I think that is what @KeXiangWang needs because we want to figure out the portion of existent prefix/key reads when not all SST bloom filters return negative (i.e. per read request true positive rate). 

For per SST bloom filter FPR, I think a micro-bench with random generated data to verify the bloom filter implementation and hashes works better than exposing a metric.